Connection is still allocated error on Windows

<!-- Please fill out the following information to help us in analyzing the issue, but feel free to skip it if you don't have the information at hand or if it does not apply. Please remove everything which does not apply to the issue. -->

### Description
d-m-p 0.28.0 is not working on Windows, e.g. `mvn docker:build`
Error: `Connection is still allocated`
0.28.0 works fine on Linux. Old version 0.27.2 works on Windows.
The issue is reproducible with d-m-p version switching back and forth, restarting, colleagues machine.
I found this issue from f-m-p, which looks related: https://github.com/fabric8io/fabric8-maven-plugin/issues/1468

```
> mvn docker:build
[INFO] Scanning for projects...
... download artifacts ...
[INFO]
[INFO] --------------------------< com.example:test >--------------------------
[INFO] Building test 1.0-SNAPSHOT
[INFO] ----------------------------[ docker-build ]----------------------------
[INFO]
[INFO] --- docker-maven-plugin:0.28.0:build (default-cli) @ test ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.089 s
[INFO] Finished at: 2019-01-17T14:55:35+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.28.0:build (default-cli) on project test: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.28.0:build failed: Connection is still allocated -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```
### Info

* d-m-p version : 0.28.0
* Maven version (`mvn -v`) : 3.5.4 or 3.6.0
* Windows version : 10.0.14393 Build 14393
* Docker version : Docker Desktop 2.0.0.2 (30215)
* How to reproduce : [windows-0.28.0.zip](https://github.com/fabric8io/docker-maven-plugin/files/2768983/windows-0.28.0.zip)

--- a/doc/changelog.md
+++ b/doc/changelog.md
@@ -3,6 +3,7 @@
 * 0.28.1
   - Reintroduce minimal API-VERSION parameter in order to support docker versions below apiVersion 1.25
   - docs: Correct default image naming
+  - close api version http connection ([#1152](https://github.com/fabric8io/docker-maven-plugin/issues/1152))
 
 * **0.28.0** (2018-12-13)
   - Update to JMockit 1.43

--- a/src/main/java/io/fabric8/maven/docker/access/hc/DockerAccessWithHcClient.java
+++ b/src/main/java/io/fabric8/maven/docker/access/hc/DockerAccessWithHcClient.java
@@ -722,8 +722,9 @@ public class DockerAccessWithHcClient implements DockerAccess {
         HttpGet get = new HttpGet(baseUrl + (baseUrl.endsWith("/") ? "" : "/") + "version");
         get.addHeader(HttpHeaders.ACCEPT, "*/*");
         get.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
-        CloseableHttpResponse response = delegate.getHttpClient().execute(get);
+        try (CloseableHttpResponse response = delegate.getHttpClient().execute(get)) {
 
-        return response.getFirstHeader("Api-Version") != null ? response.getFirstHeader("Api-Version").getValue() : API_VERSION;
+            return response.getFirstHeader("Api-Version") != null ? response.getFirstHeader("Api-Version").getValue() : API_VERSION;
+        }
     }
 }

